distributed/tests/test_client_loop.py

```diff
@@ -0,0 +1,49 @@
+import pytest
+import threading
+from time import sleep
+from distributed import LocalCluster, Client
+from distributed.utils import LoopRunner
+
+
+def run_client(loop):
+    with LocalCluster(loop=loop, asynchronous=False) as cluster:
+        client = Client(cluster, loop=loop, asynchronous=False)
+        sleep(1)  # Allow IO loop process events.
+        client.close()
+
+
+# Test if Client correctly tears down LoopRunner on close.
+@pytest.mark.parametrize("with_own_loop", [True, False])
+def test_close_loop_sync(with_own_loop):
+    loop_runner = loop = None
+
+    # Number of threads before and after the Cluster running.
+    threads_before = []
+    threads_after = []
+
+    for i in range(3):
+        threads_before.append(threading.enumerate())
+
+        if with_own_loop:
+            loop_runner = LoopRunner(asynchronous=False)
+            loop_runner.start()
+            loop = loop_runner.loop
+
+        run_client(loop)
+
+        if with_own_loop:
+            loop_runner.stop()
+
+        # Allow IO loop process events and wait until daemon threads finished.
+        sleep(1)
+        threads_after.append(threading.enumerate())
```

Comment:
It looks like as-is this test takes something like 10s.  Is there a way to reduce this?  Most tests that we run take less than a second.  We've found that as the test suite gets longer people run it less while developing.

Reply:
I'll improve, seems better to to replace `sleep`-way with  async test.